src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx

```diff
@@ -94,6 +95,12 @@ export function AccessCodeDetails({
     )
   }, [accessCode, deleteCode, onDelete, preventDefaultOnDelete])
 
+  const { device } = useDevice({ device_id: accessCode?.device_id })
+  const cannotSpecifyPinCode =
+    device?.properties.code_constraints?.some(
+      ({ constraint_type: type }) => type === 'cannot_specify_pin_code'
+    ) ?? false
```

Comment:
Minor, but the double negative checks ("if not cannot foo...") are harder to grok
```suggestion
  const canSpecifyPinCode =
    (!device?.properties.code_constraints?.some(
      ({ constraint_type: type }) => type === 'cannot_specify_pin_code'
    )) ?? true
```

Reply:
updated